### PR TITLE
Adds the Hivebot Invasion event

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Hivebots have been sighted in the space around Space Station 13; be wary of encounters.
/:cl:

* Adds a Hivebot Invasion event with a very low weight (1) that can only occur with 25 or more players.
* The crew will receive a warning 30 seconds prior to the invasion and another one 15 seconds after the first that tells where the invasion will commence.
  * This "invasion point" is told to active admins as soon as the first warning is made. It is assigned by randomly choosing from generic event spawns.
* When the invasion begins, an invasion fabricator will slam into the hull and cause a very large, light explosion.
  * The fabricator spits out hivebots at a very fast rate (around one every two seconds).
  * It can create normal, ranged, or fast-moving hivebots.
  * Immune to explosions because of how its spawning works.
* Hivebot beacons will occasionally warp in near the fabricator.
  * Spawned randomly, but chance to spawn increases every tick one isn't spawned.
* The invasion ends after 3 minutes. The fabricator may be destroyed before then, but beacons will continue to warp in until the event finishes.
* Also adds a false alarm event with the initial warning and nothing else.
  * This has a higher weight, at 5.

I made this because hivebots have existed in the code for a while and were woefully underused. I thought that this might be a fun thing to shake up rounds and make use of them as the kill-'em-with-numbers swarmers they were designed to be but never implemented.

### CODE REVIEW

**Comment by Xhuis** [comment]
Also, I think someone had a bunch of hivebot sprites on the forums. If someone finds that thread I would be happy to implement them.

**Comment by optimumtact** [comment]
another round ending "event"

**Comment by Xhuis** [comment]
What can I say? I like making big flashy stuff.

**Comment by jughu** [comment]
Ranged sounds kinda overkill at the rate it produces

**Comment by Xhuis** [comment]
I figured it would be balanced out by the fact that they have 15 health and are probably going to be restricted to a tight space.

**Comment by TehZombehz** [comment]
Sounds like a pretty extreme event, but I'd rather see how it functions in practice than assume how it'll function by just reading. The 25 player requirement is also nice.

**Comment by Xhuis** [comment]
I think it's probably the most aggressive event that the game will have if it gets merged. Others might have more lasting consequences that are more severe in the end (like space carp) but this one is just big, in-your-face invasion.

**Comment by jughu** [comment]
Eh just drop a maxcap on it everything be fine 😄

**Comment by Xhuis** [comment]
Hate to break it to you buddy, but they're immune to explosions...

**Comment by jughu** [comment]
nerf pls

**Comment by Niknakflak** [comment]
:-1:

**Comment by Core0verload** [comment]
We don't need more round ending events. At all.

**Comment by Armhulen** [comment]
I made tons of hivebot skins for an idea I had where hivebots could have dextrous hands but that all went down the shitter so unno man

**Comment by Core0verload** [comment]
Simple animals can have dexterous hands now. Might as well redo your old idea. Can't be any shittier than this.

**Comment by Xhuis** [comment]
Stay classy, Core.

**Comment by Armhulen** [comment]
if you can revive the hivebot dream then by all means my friendo, i'll sprite it all up again

**Comment by Xhuis** [comment]
I don't really know if it's better as an event, gamemode, miniantag, whatever. This is a quick attempt to make something out of them.

**Comment by Armhulen** [comment]
the event is pretty nice, i was thinking of giving them dextrous so they could be viable in xenobio again but this event is, like i said, pretty damn nice

**Comment by Kromgar** [comment]
Hivebots as a mode could be interesting sorta like blob. Have a core coordinating  brainless hivebots and the ability to make "intelligent" ones

**Comment by Armhulen** [comment]
R T S BOTS

**Comment by Core0verload** [comment]
>Stay classy, Core.

Sick and tired of round ending event bullshit. I added pop caps to events already, but the cancer of swarmers still wrecks enough rounds.

**Comment by Armhulen** [comment]
swarmers is not fun at all for other people but this ends after three minutes. It sounds maybe even a little bit fixable

**Comment by PKPenguin321** [comment]
doesn't seem round ending at all, just a thing that spawns some obnoxious mobs for 3 minutes before dying

**Comment by Armhulen** [comment]
reminder most hivebots can be killed in one hit by a welder

**Comment by FantasticFwoosh** [comment]
Dropping the hivebot replicator by itself as a 'gateway' for ghost joins would be better, but without defined goals that just resorts to murder-bone since players would be more handy with infinite bullets than a mob.

**Comment by ExcessiveUseOfCobblestone** [comment]
R[ound] E[nding] E[v]E[nts]

**Comment by optimumtact** [comment]
I don't like this @tgstation/commit-access someone else decide

**Comment by AnturK** [comment]
Sounds alright but throw away that randomly spawning nuke-ops then, this is bit more fleshed out.

**Comment by Armhulen** [comment]
https://www.dropbox.com/s/820yj8f661xndn0/newbots.dmi?dl=0

i'm too lazy to add this myself but xhius if you want to feel completely free to do so

((((((((((((((((((please don't add dsbot)))))))))))))))))

**Comment by KorPhaeron** [comment]
Agree with Anturk

**Comment by Xhuis** [comment]
I'll disable it once I get home

**Comment by FantasticFwoosh** [comment]
Warp storm nuke ops spews traitor loot everywhere just by easily baiting & abusing the ops pathing, wheras a anomaly core dropped by the hivebot-replicator or random level 2/3 R&D parts from fallen hivebots would be a nice incentive besides being murdered to justify a armed response for a lootgrab if they arent too strong.

You can sell the parts off in bulk if you aren't using them or scrap them in a recycler for metal/glass content.

**Comment by Xhuis** [comment]
Actually, hivebots don't drop any actual loot. I should give the invasion dispenser something nice to give some incentive to actually destroy it rather than sanction off the area and let the server die of lag.

**Comment by Xhuis** [comment]
@AnturK By "randomly spawning nuke ops", do you mean the solo nuke op event (is that still around?) or portal storms?

**Comment by Xhuis** [comment]
May or may not be in the process of giving the fabricator a giant buzzsaw

**Comment by Armhulen** [comment]
kek!

have you considered the strong hivebot skin?

**Comment by Xhuis** [comment]
Made a lot of changes!
* Replaced the invasion fabricator with the [hivebot swarm core](http://puu.sh/tHDNd/1fd68bf7f7.png).
  * The hivebot swarm core makes a hivebot every tick while not in combat (up to 50 hivebots in a 7x7 square) and has 250 health.
  * It locks on to carbon creatures in its view (not silicons, since it's too dumb to recognize bot from bot) and will attack its lock-on until they're far away, the creature dies, or it dies
  * 2/3 of the time, it will fire a beam that deals 15 burn damage
  * 1/3 of the time, it will rev its buzzsaw and slam it down onto the tile its target was standing on when the charge started; anyone standing there or who moves there during the attack takes massive brute damage over two seconds
  * Probably more attacks to come
  * Drops the __hivebot crux__
* Added the hivebot crux
  * Summons a friendly, non-combat hivebot pet to be generally cute and help you with stuff
  * Nothing yet but it's planned to do things like fetch objects and toss stuff into disposal units
* Beacons no longer spawn around the invasion point, to make up for the swarm core's defenses

**Comment by Xhuis** [comment]
Still undecided about where to go with the crux. I might just rework it into something entirely different.

**Comment by Armhulen** [comment]
This thing is actually extremely weak. IT DOESNT SPAWN MORE HIVEBOTS IF YOURE IN RANGE? gamesux, 15 burn lasers not only will confuse balance as almost all normal lasers do 20. It also kinda makes me sad considering hivebots have guns and claws and you just randomly stick on a laser? I dunno, that's what I think about it

**Comment by Armhulen** [comment]
And only give the special item if they kill it. Why even attack it if you can just wait 3 mins and get free loot anyways.

**Comment by Armhulen** [comment]
AND, final comment, I think they should totally be payday style land on station, drain funds for three mins and then if you haven't killed the bots in time they all ( remaining anyways) get back on board and blast off into space. But that's just a suggestion to take them in instead of random griffbots

**Comment by Xhuis** [comment]
* It's very fearsome if you're alone, but more attacks are coming eventually. As for lasers, it's what the sprite had, so I rolled with it.
* The crux is a _drop_. People can't get it by waiting it out, and with the swarm core, it won't ever despawn.

**Comment by Armhulen** [comment]
it doesn't leave? why? this isn't supposed to leave but if it's not attacked quickly enough, it'll spawn 50 hivebots and just shut down that area of the station, not to mention that, right, you get a warning, which now makes it a pseudo round ending event, where you either get to kill it ten seconds in or it's there to stay.

That being said x-ray would kick it's ass and i do like counters. I liked the coming (antagonizing, see moneystealing idea) going idea, as it means that it's not forcefully murder-ish and it's not forcefully round ending. If nobody is prepared to instantly destroy the swarm core then people would rather just call it then wait for x-rays, and that's even if you have scientists who are competent and remember to counter it.

tl;dr I loved the original idea of come fuck shit up and then go because it's COMPLETELY optional to fight it whereas this makes it either a free shuttle call or free loot.

also: I could look into making a better themed swarm core, to give it that hivebot shine. you know, immershuns.

**Comment by Xhuis** [comment]
You could if you'd really like. The current sprite fits fine, and I can still make it work. As for if/when it leaves, I can put that in, as it isn't set in stone. Even in the original event, the fabricator stayed; beacons just kept spawning in and stopped at a certain point. The fabber didn't _have_ a hivebot limit so it could go and go and go until the server crashed.

**Comment by Armhulen** [comment]
All those changes are great, I might start work on a swarm core

**Comment by Xhuis** [comment]
Up to you. I'll send you what I'm using right now so you can see what you want to replace, or just download the dmi from the branch.

**Comment by Xhuis** [comment]
Should've probably included that I'll do that when I get home in a few hours. Oops.

**Comment by Xhuis** [comment]
Still recovering from my repo being corrupted so here's the old screenshot. It's a little more animated but this is the general gist.

https://puu.sh/tHDNd/1fd68bf7f7.png

**Comment by Steelpoint** [comment]
Years ago I proposed a Hivebot game mode where you had player controlled Hivebots attempting to take over the station, each commanding squads of Hivebots as well as there being free roaming Hivebots. This helped give a sense of a true invasion. Shame there's no archive of the old Wildhogs forum.

But I digress I think this is a interesting feature to add. I always liked Hivebots because you can send in a swarm of them and even the most unprepared of players can still fight and even come out on top of a small group of Hivebots. Thus it can really simulate a invasion.

**Comment by Armhulen** [comment]
I am back to being really hyped

**Comment by Xhuis** [comment]
@Steelpoint A gamemode would be more fitting for hivebots. We have the assets. I feel like with LeoZ doing a rework of gamemode code and me working on umbrages, though, that it's not worth my effort.

**Comment by Xhuis** [comment]
This is turning into a pretty big project for me, and I don't want to have two big projects at once (the other being umbrage) so I'm gonna close this for now.
